Add dropdown support for Sphinx Design

choldgraf opened the pull request:
This adds Dropdown support for Sphinx Design. It re-works the way color variables are defined a bit, so that they are at the top of the page and thus can be re-used inside it.

If a color is specified, it makes Sphinx Design dropdowns behave similarly to our admonitions, with a title background shaded w/ the color, and a border to the left.

Here's how they look in light / dark mode:

![chrome_qOQv2PM6Q9](https://user-images.githubusercontent.com/1839645/179195733-976adf2e-c7de-4851-9a38-8fcc81848920.gif)

drammock commented:
@choldgraf could you self-review to point out where the important / new parts of this are?  There seem to be big chunks that were just moved within the file, and I'm having a bit of trouble seeing what needs scrutiny here.  Alternatively you could rebase/force-push to separate the purely-move changes into one commit, and the substantive changes in another commit.

choldgraf commented:
I think really the only structural things that I changed were to move the colors and variables part of that sphinx design file to the top instead of the very bottom, so that I can re-use the sphinx color variables in that page.

choldgraf commented:
If it's confusing let me know and I will give it a shot

drammock commented on src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss:
```diff
@@ -78,71 +144,55 @@ html[data-theme="light"] {
-$all-colors: map-merge($pst-semantic-colors, $extra-semantic-colors);
+  // Drop shadow should behave same as admonitions
+  @include box-shadow();
+  summary.sd-card-header {
+    position: relative; // So background color works
+    font-weight: 600;
+    padding-top: 0.5rem;
+    padding-bottom: 0.5rem;
+    border-left: 1px solid var(--sd-color-card-border);
```
just wondering how these were chosen (and whether it makes sense to have CSS variables for them?)...  anytime we're writing style rules to "make this component look like this other component" it makes me wonder

choldgraf replied:
I think in this case I am just literally copying the border rule that sphinx design uses. The problem was that we are putting the border on the title and content blocks, while sphinx design was putting it on the whole details block only when it was closed, and putting it on the title only when it was open, or something like that.

Maybe I should add a comment to explain that?

drammock commented on src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss:
```diff
@@ -78,71 +144,55 @@ html[data-theme="light"] {
 
+/* Finally some manual UI element replacements.
+ */
 // Now we override the --sd-color-* variables.
```
```suggestion
```

drammock reviewed (approved):
looks reasonable. one question and one cruft-comment removal suggestion

choldgraf commented:
OK I believe I've addressed all comments above, and have also added a bunch of other code comments to try and make the colors / dropdowns logic more understandable. Let me know how this looks.

drammock reviewed (approved):
thanks for the additional comments! very helpful

choldgraf commented:
I'm going to merge this one in so that I can try applying some similar stuff to fix #817
